connector-sdk/core/src/main/java/io/camunda/connector/api/inbound/CorrelationFailureHandlingStrategy.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.api.inbound;
+
+/**
+ * Strategy to handle correlation failures. The Connector runtime provides this strategy to the
+ * connector implementation as a hint on how to handle the failure. The connector implementation is
+ * ultimately not required to follow this strategy, but it is a good practice to do so.
+ *
+ * <p>The Connector runtime may provide this information based on:
+ *
+ * <ul>
+ *   <li>the type of the failure
+ *   <li>the configuration of the connector provided by the user
+ *   <li>the configuration of the Connector runtime
+ * </ul>
+ */
+public sealed interface CorrelationFailureHandlingStrategy {
+
+  /**
+   * When the connector implementation receives this strategy from the runtime, it should forward
```

Comment:
If we don't have any docs, it would nice to add an example. Maybe the initial issue we had with the RabbitMQ connector?

Reply:
this example should probably on the `correlate` method as this is what the user will use

Reply:
I added this info to the `correlate` method as well 👌